services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/MetersValidateSimpleFlowSpec.groovy

```diff
@@ -0,0 +1,517 @@
+
+import groovy.transform.Memoized
+import org.springframework.beans.factory.annotation.Autowired
+import spock.lang.Narrative
+
+@Narrative("""This test suite check the meter validate request on a flow that contains 2+ switches.
+description of fields:
+- missing - those meters/rules, which is NOT exist on a switch, but it is exist in db
+- misconfigured - those meters which have different value (on a switch and in db) for the same parameter
+- excess - those meters/rules, which is exist on a switch, but it is NOT exist in db
+- proper - meters/rules value is the same on a switch and in db
+""")
+class MetersValidateSimpleFlowSpec extends BaseSpecification {
+    @Autowired
+    SwitchHelper switchHelper
+
+    def "Able to get meter validate information"() {
+        when: "Create a flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def defaultMetersSrcSwitch = northbound.getAllMeters(srcSwitch.dpId)
+        def defaultMetersDstSwitch = northbound.getAllMeters(dstSwitch.dpId)
+        def flow = flowHelper.addFlow(flowHelper.randomFlow(srcSwitch, dstSwitch))
+
+        then: "One meter is automatically created on the both switches"
+        def createdMetersIdSrcSwitch = getCreatedMetersId(srcSwitch.dpId, defaultMetersSrcSwitch)
+        def createdMetersIdDstSwitch = getCreatedMetersId(dstSwitch.dpId, defaultMetersDstSwitch)
+        def burstSizeSrcSwitch = switchHelper.getExpectedBurst(srcSwitch.dpId, flow.maximumBandwidth)
+        def burstSizeDstSwitch = switchHelper.getExpectedBurst(dstSwitch.dpId, flow.maximumBandwidth)
+
+        createdMetersIdSrcSwitch.size() == 1
+        createdMetersIdDstSwitch.size() == 1
+
+        and: "The correct info is stored in the validate meter response"
+        def meterValidateInfoSrcSwitch = northbound.validateMeters(srcSwitch.dpId)
+        def meterValidateInfoDstSwitch = northbound.validateMeters(dstSwitch.dpId)
+        def createdCookieSrcSwitch = getCreatedCookies(srcSwitch.dpId, createdMetersIdSrcSwitch)
+        def createdCookieDstSwitch = getCreatedCookies(dstSwitch.dpId, createdMetersIdDstSwitch)
+
+        def createdCookies = createdCookieSrcSwitch + createdCookieDstSwitch
+
+        meterValidateInfoSrcSwitch.meters.proper.collect { it.meterId }.containsAll(createdMetersIdSrcSwitch)
+        meterValidateInfoDstSwitch.meters.proper.collect { it.meterId }.containsAll(createdMetersIdDstSwitch)
+
+        meterValidateInfoSrcSwitch.meters.proper.collect { it.cookie }.containsAll(createdCookieSrcSwitch)
+        meterValidateInfoDstSwitch.meters.proper.collect { it.cookie }.containsAll(createdCookieDstSwitch)
+
+        [meterValidateInfoSrcSwitch, meterValidateInfoDstSwitch].each {
+            it.meters.proper.each {
+                assert it.rate == flow.maximumBandwidth
+                assert it.flowId == flow.id
+            }
+        }
+
+        meterValidateInfoSrcSwitch.meters.proper.burstSize[0] == burstSizeSrcSwitch
```

Comment:
why you have to take `[0]` from the burstSize? Didn't quite get why it's a collection and why we verify only first value

Reply:
Fixed